src/pages/home/index.js

```diff
@@ -0,0 +1,582 @@
+
+  /** MISC **/
+  const [date, setDate] = React.useState(new Date());
+
+  /** PAGINATION **/
+  const [currentPage, setCurrentPage] = React.useState(1);
+  const [vaccinePerPage] = React.useState(3);
+  const indexOfLastVaccine = currentPage * vaccinePerPage;
+  const indexOfFirstVaccine = indexOfLastVaccine - vaccinePerPage;
+  const [pages, setPages] = React.useState([]);
+
+  React.useEffect(() => {
+    if (vaccines && vaccines.length) {
+      setCurrentVaccines(
+        vaccines.slice(indexOfFirstVaccine, indexOfLastVaccine)
+      );
+      for (let i = 1; i <= Math.ceil(vaccines.length / vaccinePerPage); i++) {
+        setPages((old) => [...old, i]);
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [vaccines, currentPage]);
+
+  const paginate = (pageNumber) => {
+    setCurrentPage(pageNumber);
+  };
+
+  const nextPage = () => {
+    setCurrentPage(currentPage + 1);
+
+    if (currentPage + 1 > pages.length) {
+      setCurrentPage(pages.length);
+    }
+  };
+
+  const prevPage = () => {
+    setCurrentPage(currentPage - 1);
+
+    if (currentPage - 1 <= 0 && pages.length) {
+      setCurrentPage(pages.slice(0, 1));
+    }
+  };
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  React.useEffect(() => {
+    if (!states) {
+      if (/district/gi.test(selectedCriteria)) {
+        getStates();
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedCriteria]);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  React.useEffect(() => {
+    getDistricts();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedState]);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  React.useEffect(() => {
+    if (selectedDistrict || selectedPin) {
+      getVaccines();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedDistrict, selectedPin, date, selectedCriteria]);
+
+  const generateURL = (
+    _sub = null,
+    _pin = null,
+    _district = null,
+    _calendar = false
+  ) => {
+    if (_sub) {
+      return `${BASE_URL}${_sub}`;
+    }
+
+    let sub;
+    const formattedDate = typeof date === "string" ? date : formatDate(date);
+
+    if (_pin && _calendar) {
+      sub = `/appointment/sessions/public/calendarByPin?pincode=${_pin}&date=${formattedDate}`;
+    } else if (_pin) {
+      sub = `/appointment/sessions/public/findByPin?pincode=${_pin}&date=${formattedDate}`;
+    } else if (_district && _calendar) {
+      sub = `/appointment/sessions/public/calendarByDistrict?district_id=${_district.district_id}&date=${formattedDate}`;
+    } else if (_district) {
+      sub = `/appointment/sessions/public/findByDistrict?district_id=${_district.district_id}&date=${formattedDate}`;
+    }
+
+    const url = `${BASE_URL}${sub}`;
+    return url;
+  };
+
+  const getStates = async () => {
+    try {
+      setLoadingStates(true);
```

Comment:
Can you put this loading state as always true and only evaluates to false only when the states are fetched, in that case you dont have to call it twice one at line 158 and one at 165.

Reply:
@stephin007, that is doable but not considered a best practice. Usually, you want the state to be as generic as possible. Also, on refetch, you'll still have to set it to true first anyways.